app/models/user_role.rb

```diff
@@ -88,12 +89,15 @@ module Flags
   attr_writer :current_account
 
   validates :name, presence: true, unless: :everyone?
-  validates :color, format: { with: /\A#?(?:[A-F0-9]{3}){1,2}\z/i }, unless: -> { color.blank? }
+  validates :color, format: { with: VALID_COLOR }, if: :color?
 
-  validate :validate_permissions_elevation
-  validate :validate_position_elevation
-  validate :validate_dangerous_permissions
-  validate :validate_own_role_edition
+  with_options if: -> { defined?(@current_account) } do
```

Comment:
It's always satisfying to see the "when do we run this" conditions moved up to the call site and the "is this valid" portion left down in the method, and I want to just thank you for doing that here.